### PR TITLE
AAE-22900: update ObjectDataTableAdapter to support server and client…

### PR DESCRIPTION
**Pull request by amolodyh-hyland**
Fixes [AAE-22900](https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiYmU2M2FhMzFmMjFmNGVjNDkxOGRmMmNhOGI2MGE5Y2YiLCJwIjoiaiJ9). This adds server sorting for the base `ObjectDataTableAdapter` to enable tables in tabs such as "Process Admin" User Tasks, Services Tasks, and Applications List.

**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
 [AAE-22900](https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiYmU2M2FhMzFmMjFmNGVjNDkxOGRmMmNhOGI2MGE5Y2YiLCJwIjoiaiJ9)

**What is the new behaviour?**

Allows for enabling server sorting where it the adapter does not sort the data. The server data is rendered in the order it comes in.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

This a partial fix to the provided story  [AAE-22900](https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiYmU2M2FhMzFmMjFmNGVjNDkxOGRmMmNhOGI2MGE5Y2YiLCJwIjoiaiJ9). After this is merged, hxp-frontend-apps will need to be updated by updating the `applications-list-filtered-data-adapter` adapter as well.

[AAE-22900]: https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AAE-22900]: https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/process-services-cloud/src/lib/process/process-list/datatable/process-list-datatable-adapter.ts
Comment (amolodyh-hyland) [inline lib/process-services-cloud/src/lib/process/process-list/datatable/process-list-datatable-adapter.ts:23]: @DenysVuika @eromano Do we want to provide the ability to set the `sortingMode` in these extending classes so we can utilize feature flags in the hxp-frontend-apps projects? Not sure if we want a feature flag for this but if we did, we'd need to be able to provide the sort option here.

Based on the information I've received, we don't want pages using these adapters to use client sorting so the constructor parameter would be redundant. Also, we would have to remove the constructor parameter once the feature flag is removed so it seems architecturally redundant.

File: lib/process-services-cloud/src/lib/task/task-list/datatable/task-list-datatable-adapter.ts
Comment (amolodyh-hyland) [inline lib/process-services-cloud/src/lib/task/task-list/datatable/task-list-datatable-adapter.ts:23]: @DenysVuika @eromano Same question for this as the other adapter class comment.

File: lib/core/src/lib/datatable/data/object-datatable-adapter.ts
Comment (DenysVuika) [inline lib/core/src/lib/datatable/data/object-datatable-adapter.ts:64]: this might be a breaking change if components rely on the client sorting by default... needs to be somehow verified with ACA/ADW probably. cc @MichalKinas

File: lib/process-services-cloud/src/lib/process/process-list/datatable/process-list-datatable-adapter.spec.ts
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/process/process-list/datatable/process-list-datatable-adapter.spec.ts:56]: the test should not be touching private methods... I suggest making a public accessor to the sorting mode (indexed access to private properties is configured a hack or the temp limitations of TS)

Comment (amolodyh-hyland) [reply lib/process-services-cloud/src/lib/process/process-list/datatable/process-list-datatable-adapter.spec.ts:56]: Yeah, I'm not a fan of accessing private variables either but didn't want to create a public field just for the sole purpose of testing. I created tests to verify the sort direction of the data but that triggered duplicate code warnings in the PR.

I can expose the field publicly if that's the suggested patter for this type of scenario?

File: lib/core/src/lib/datatable/data/object-datatable-adapter.ts
Comment (amolodyh-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts:64]: @DenysVuika This shouldn't break sorting for any components that don't explicitly set the sortingMode to `'server'`. From my understanding, currently this will only affect the task and process list adapters.

File: lib/process-services-cloud/src/lib/process/process-list/datatable/process-list-datatable-adapter.spec.ts
Comment (DenysVuika) [reply lib/process-services-cloud/src/lib/process/process-list/datatable/process-list-datatable-adapter.spec.ts:56]: the way you access private is more of a hack, any time MS fixes the TS and everything may start failing... or it will me marked as security issue... it's better either not touching the private, or testing other code that involves this private, or making an accessor

File: lib/core/src/lib/datatable/data/object-datatable-adapter.ts
Comment (MichalKinas) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts:64]: This not entirely true since at least search results page in ACA is using the 'server' mode and it might be set through the user preferences also for other lists as far as I can see, @amolodyh-hyland before merging the PR please create linked PR in ACA to see if all tests will pass

Comment (amolodyh-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts:64]: @MichalKinas Could you point me to which search results you're referring to? I just want to take a look at what adapter it's using.

Comment (amolodyh-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts:64]: @MichalKinas If you're referring to the document list table then that one uses its own implementation and this update wont affect it.

Comment (MichalKinas) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts]: Yes it's about the table in the document list

Comment (amolodyh-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts]: @MichalKinas Should I still run a branch with link-adf since the documents list doesn't use the `ObjectDataTableAdapter`?

Comment (MichalKinas) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts]: That's great but this is a PR in HX monorepo and Denys wanted to check https://github.com/Alfresco/alfresco-content-app which is referred to as ACA while adding the comment

Comment (amolodyh-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts]: Ah, I see. Wasn't aware of that repo. I'll get the build ran over there as well then.

Comment (amolodyh-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts]: @MichalKinas I created a PR with a linked commit and everything passed here: https://github.com/Alfresco/alfresco-content-app/pull/4278

However, I'm not 100% sure if that's the correct way to link the ADF library to the ACA PR. Is that the right way to achieve this?

Comment (MichalKinas) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts]: The commit message was correct but for some reason ADF was not linked, I pushed new empty commit to your branch and now it seemed to work which you can observe here: https://github.com/Alfresco/alfresco-content-app/actions/runs/12160366230/job/33912497738?pr=4278#step:4:269, lets see if PR will pass now

Comment (amolodyh-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts]: Looks like the build succeeded, minus the "Finalize" step. The finalize step looks like it failed because of the adf-link commit (I'm assuming this is expected?).

Comment (MichalKinas) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts]: Yes this is expected, all good on ACA side

**Review by MichalKinas — APPROVED** [review]

File: lib/core/src/lib/datatable/data/object-datatable-adapter.ts
Comment (MichalKinas) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts]: Looks good to me, when Denys approves you can merge it

**Review by eromano — APPROVED** [review]

**Comment by eromano** [comment]
@amolodyh-hyland you van squash merge when you are ready and is green if the PR should not merged lease add draft or do not merge

**Pull request merged**
